src/Model/Table/VocabularyTable.php

```diff
@@ -125,15 +125,13 @@ private function _getNumberOfSentences($lang, $text)
         if (!Configure::read('Search.enabled')) {
             return null;
         }
-        $index = array($lang . '_main_index', $lang . '_delta_index');
-        $sphinx = array(
-            'index' => $index,
-            'matchMode' => SPH_MATCH_EXTENDED2
-        );
-        $query = Search::exactSearchQuery($text);
+        $search = new Search();
+        $search->filterByLanguage($lang);
+        $search->filterByQuery(Search::exactSearchQuery($text));
+        $search->sort('random');
```

Comment:
Why randomize the results?

Reply:
This is a workaround to set the `SPH_MATCH_EXTENDED2` match mode. Without it the default match mode would be `SPH_MATCH_ALL`.
So if there are two sentences `He goes.` and `She goes and he runs.`, on the vocabulary page there will be 2 sentences found for the vocabulary item `he goes`. But after clicking on the link, there will only be one sentence found because the search page uses the extended mode.
(Since we are only interested in the count the order isn't relevant.)

Should I add a new method to the `Search` model which allows to set the match mode independent of the sorting? Or should I use another sorting order?

Reply:
I see! For now, the Search model implementation kinda hides the match mode thing, so I think we can set it to `SPH_MATCH_EXTENDED2` regardless of `Search->sort` being set or not. From the caller’s perspective, it doesn’t make sense to filter exact matches with `="..."` and actually not get exact matches in the results.